Programming Foundations: Algorithms/Factorial_test/lib/fact.rb

```diff
@@ -0,0 +1,22 @@
+module Factorial_test
+  class Fact
+    def initialize(n)
+      @n = n
+    end
+
+    def fact
+      if @n == 0
+        1
+      elsif @n < 0
+        raise 'negatives are not acceptable'
+      else
+        result = 1
+        while @n > 0
```

Comment:
As stateed by @Martouta, when you have `Enumerable`, try to read the docs and use some of the methods of this module.

Try to understand this simpler version is:

```ruby
(1..n).inject(&:*)
```